src/cfn/convertStackToIidy.ts

```diff
@@ -0,0 +1,213 @@
+    return sortMapByWeights(node0, iamStatementSortWeights);
+  } else if (_.includes(['PolicyDocument', 'AssumeRolePolicyDocument'], currentNode)) {
+    return sortMapByWeights(node0, {'Version': 0, 'Statement': 1});
+  } else if (parentNode === 'Policies' && _.isObject(node0)) {
+    return sortMapByWeights(node0, {'PolicyName': 0, 'PolicyDocument': 1});
+  } else {
+    return node0;
+  }
+}
+
+function deepMapValues(object: any, callback: any, propertyPath: any): any {
+  if (_.isArray(object)) {
+    return _.map(object, deepMapValuesIteratee);
+  } else if (object instanceof yaml.Tag) {
+    return object.update(deepMapValues(object.data, callback, propertyPath));
+  } else if (_.isObject(object) && !_.isDate(object) && !_.isRegExp(object) && !_.isFunction(object)) {
+    return _.extend({}, _.mapValues(callback(object, propertyPath), deepMapValuesIteratee));
+  } else {
+    return callback(object, propertyPath);
+  }
+
+  function deepMapValuesIteratee(value: any, key: string) {
+    var valuePath = propertyPath ? propertyPath + '.' + key : key;
+    return deepMapValues(value, callback, valuePath);
+  }
+}
+
+const DEFAULT_SORT_WEIGHT = 9999;
+function sortMapByWeights(node0: any, weights: any): any {
+  let nodePairs = _.sortBy(_.toPairs(node0));
+  nodePairs = _.sortBy(nodePairs, ([k, v]) => _.get(weights, k, DEFAULT_SORT_WEIGHT));
+  return _.fromPairs(nodePairs);
+}
+
+export async function convertStackToIIDY(argv: Arguments): Promise<number> {
+  await configureAWS(argv.profile, argv.region);
+  const outputDir = argv.outputDir;
+  const StackName = argv.stackname;
+  const TemplateStage = def('Original', argv.stage);
+
+  const cfn = new aws.CloudFormation();
+  const {TemplateBody} = await cfn.getTemplate({StackName, TemplateStage: 'Original'}).promise();
+  if (!TemplateBody) {
+    throw new Error(`Invalid cfn template found for ${StackName}`);
+  }
+
+  const templateObj0 = parseTemplateBody(TemplateBody!);
+  let templateObj;
+  if (argv.sortkeys) {
+    templateObj = deepMapValues(templateObj0, sortCallback, 'root');
+  } else {
+    templateObj = templateObj0;
+  }
+
+  const stack = await getStackDescription(StackName);
+  if (!fs.existsSync(outputDir)) {
+    fs.mkdirSync(outputDir);
+  }
+  const {StackPolicyBody} = await cfn.getStackPolicy({StackName}).promise();
+  let StackPolicy: object;
+  if (StackPolicyBody) {
+    StackPolicy = JSON.parse(StackPolicyBody);
+  } else {
+    StackPolicy = {
+      "Statement": [
+        {
+          "Effect": "Allow",
+          "Action": "Update:*",
+          "Principal": "*",
+          "Resource": "*"
+        }
+      ]
+    };
+  }
+  fs.writeFileSync(pathmod.join(outputDir, 'stack-policy.json'), JSON.stringify(StackPolicy, null, ' '));
+  const originalFileExt = (TemplateBody.match(/^ *\{/) !== null) ? 'json' : 'yaml';
+  fs.writeFileSync(pathmod.join(outputDir, '_original-template.' + originalFileExt), TemplateBody);
```

Comment:
Unrelated to these changes, but have you considered just printing this to stdout rather than creating a file for it? I have seen there has been some confusion on whether this file needs to be kept around or not.

Reply:
`_original-template.ext`

Reply:
We could change the filename to clarify that. `_original-template.backup.ext`